### PR TITLE
Batch show with items

### LINKED ISSUE #76: Batch-Show component
We have the UX for BatchShow:
image

Note these details:
- two tabs: One to view/edit details, the other for items in this batch
- "Muster List" button creates a single-page report like this:
    - Header containing Batch Number, created_at, Project name, platform name
    - a list of items, with these columns
        - item_number
        - mediaType
        - vaultLocation
- For "Change location", user will select one or more items, then click "Change Location". A popup dialog will let the user choose a new `Vault Location` for these items

### PR DESCRIPTION
**Pull request by salony75way**
Batch show with items
fixes #76

### CODE REVIEW

**Comment by IanMayo** [comment]
Codacy has a couple of issues for you to fix, please @salony75way

**Comment by IanMayo** [comment]
Could we have these changes please?
- lay out the `detail` tab like we do for `Batch Create`
- move `Create item` into the toolbar for `ItemList`. Change to `ADD ITEM`
- for `Batch List` change `CREATE` label to `ADD NEW BATCH`
- creating the `Muster List` as `.csv` is clever, but they are going to print it off, check the items, sign it, and file it away.  So, we need to be in more control of the layout - I think we need it as a web-page, please.  We'll do the detailed layout when I get a copy of the current form.

**Comment by salony75way** [comment]
> Thanks @salony75way - I _think_ we just have item 1 to sort - the layout of the details tab in `BatchShow`

Yes, First one is still pending

**Comment by IanMayo** [comment]
That's some good code there @salony75way - well done.

We need confirm when deleting items from the `items` tab of the `BatchShow` page.

File: src/resources/items/ItemList.tsx [hunk 1]
```diff
@@ -61,9 +62,16 @@ const ItemActions = () => (
   </TopToolbar>
 )
 
-export default function ItemList(): React.ReactElement {
+export default function ItemList(
+  props?: Omit<ListProps, 'children'>
```
[the remark this answers is not in the document]
Comment (salony75way) [reply src/resources/items/ItemList.tsx:66]: Hi @IanMayo, 
We are already using children in item list like this `<ItemList>children</ItemList>`
So children as a prop will not works here as children will be overwritten.

[the remark this answers is not in the document]
Comment (salony75way) [reply src/resources/items/ItemList.tsx:66]: By default `DatagridConfigurable` component is children in ItemList, And if we want to allow overwrite this from outside so we need to write some additional code for it (for example you can check `Printable` component) and we have to call ItemList component like `<ItemList>some children data<ItemList>` (as we did for `Printable` component).

So in this case we don't need to overwrite ItemList children and also children was required in `ListProps` so i omitted this.

Screenshot 2023-04-13 at 2 25 24 PM

[the remark this answers is not in the document]
Comment (IanMayo) [reply src/resources/items/ItemList.tsx:66]: Great, I see.  Thanks for taking the time to explain that :-D

**Review by IanMayo — APPROVED** [review]

**Pull request merged**